packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -166,6 +168,7 @@ export class AppManager implements BackstageApp {
   private readonly icons: NonNullable<AppOptions['icons']>;
   private readonly plugins: Set<CompatiblePlugin>;
   private readonly components: AppComponents;
+  private readonly extensions: ComponentExtensions;
```

Comment:
Would be nice to already now find a different name here tbh, perhaps to the concept as a whole. The "extension" name is currently occupied by the extensions that plugin export to make their content available to apps.

Reply:
Yeah I realised that way too late 🙈, that's an unfortunate naming collision.

Augmentation? Decoration? Addition? Customization?

Or, can we un-"extension" the current components, which are basically just components _bound_ to a plugin? Yes it'd be breaking, but could be smoothly done over time by allowing the old name too.

Reply:
We're likely about to have "extensions" become a thing in the backend system as well, with a somewhat similar role to the existing frontend extensions 😅.

Don't mean to bring the bike shed yet tbh, I think we can focus on the other parts first, just wanted to bring it up so that we can think about it a bit.

Reply:
I'ma keep bike shedding while we're on that topic. How about component "addon"?

Reply:
Yeah... `createComponentSwizzle()`, `swizzleComponent()`, `<ProvideSwizzle>`...

I don't think techdocs owns the 'addon' wording though, but maybe 'decoration' is the closest thing that somewhat accurately describes what this is.

Reply:
Just to be clear, what do the 'extension' word mean in `createComponentExtension`? What is _extending_ what there? Reading [this](https://backstage.io/docs/plugins/composability#extensions), it seems to basically be anything the plugin exports where the thing is _tied_ to the plugin.

This wording is a bit unfortunate, as it hijacks the word from where it _really_ would make sense, which is when _extending_ something. `createRoutableComponent` and `createComponent` instead of `*Extension`, or something like `exposeRoutableComponent` / `exposeComponent` would solve that. But yeah, that'd be breaking, eventually.

I'm leaning towards using _decoration_ here because of this, and piggy-back on the `.provide()` to get a uniform API.